src/serialize.ts

```diff
@@ -16,24 +16,25 @@ import { MDXRemoteSerializeResult, SerializeOptions } from './types'
  * Related: https://nextjs.org/docs/basic-features/data-fetching#reading-files-use-processcwd
  */
 function setEsbuildBinaryPath() {
-  const baseDir = pkgDir.sync() || process.cwd()
+  /**
+   * Now we have the correct path for ESBUILD (with ESBUILD_PATH), we can replace baseDir
+   */
 
+  //  const baseDir = pkgDir.sync() || process.cwd()
+  const baseDir = process.env.ESBUILD_PATH
+
+  if (!baseDir) {
```

Comment:
Had to do this validation to ensure the value is there and TS stop complains about its type being `string | undefined`